src/sentry/runner/commands/cleanup.py

```diff
@@ -332,36 +353,59 @@ def is_filtered(model: type[Model]) -> bool:
         metrics.timing("cleanup.duration", duration, instance=router, sample_rate=1.0)
         click.echo("Clean up took %s second(s)." % duration)
 
+    # Check for models that were specified but never attempted
+    if model_list:
+        # Exclude models that were legitimately filtered out (silo/router restrictions)
+        models_never_attempted = model_list - models_attempted - models_legitimately_filtered
+        if models_never_attempted:
+            logger.error(
+                "Models specified with --model were never attempted for deletion, must configure cleanup for this model",
```

Comment:
This issue was discovered: https://sentry.sentry.io/issues/6884198392/?project=1&query=is%3Aunresolved%20transaction%3A%5Bcleanup%2Csentry.deletions.tasks.run_deletion%5D&referrer=issue-stream